lib/python/picongpu/picmi/diagnostics/openpmd_sources/source.py

```diff
@@ -0,0 +1,86 @@
+"""
+This file is part of PIConGPU.
+Copyright 2021-2025 PIConGPU contributors
+Authors: Masoud Afshari
+License: GPLv3+
+"""
+
+from .bound_electron_density import BoundElectronDensity
+from .charge_density import ChargeDensity
+from .counter import Counter
+from .density import Density
+from .energy import Energy
+from .energy_density import EnergyDensity
+from .energy_density_cutoff import EnergyDensityCutoff
+from .larmor_power import LarmorPower
+from .macro_counter import MacroCounter
+from .mid_current_density_component import MidCurrentDensityComponent
+from .momentum import Momentum
+from .momentum_density import MomentumDensity
+from .weighted_velocity import WeightedVelocity
+
+import typeguard
+from typing import List, Union
+
+
+@typeguard.typechecked
+class Source:
+    """
+    Consolidates data sources for the openPMD plugin in PIConGPU.
+
+    This class aggregates individual data sources (e.g., ChargeDensity) or predefined
+    keywords (species_all, fields_all) to define the --openPMD.source parameter:
+    https://picongpu.readthedocs.io/en/latest/usage/plugins/openPMD.html
+
+    Parameters
+    ----------
+    sources: List[Union[str, <DataSourceClasses>]]
+        List of data sources, either as strings (e.g., "species_all", "fields_all") or
+        as data source objects (e.g., ChargeDensity, Density).
+    """
```

Comment:
@mafshari64 could you please comment on why you consider the aggregator functionality class necessary? I currently don' see why it is needed, it looks like the entire functionality could be moved to the openPMD plugin.
Instead I would use this class as an interface for all source implementations